pkg/codegen/schema/schema.go

```diff
@@ -535,6 +535,8 @@ type Function struct {
 	Outputs *ObjectType
 	// The return type of the function, if any.
 	ReturnType Type
+	// The return type is plain and not wrapped in an Output. This is only supported for resources.
```

Comment:
FunctionSpec unmarshal code must need updating from this, otherwise this won't actually generate correctly for Go/Node SDKs.

Reply:
Can you elaborate? The tests are verifying it working end to end.

bindFunctionDef sets ReturnTypePlain when producing *Function:

[[file:~/code/pulumi/pkg/codegen/schema/bind.go::func (t *types) bindFunctionDef(token string) (*Function, hcl.Diagnostics, error) {]]

In can see custom marshal code like unmarshalFunctionSpec but it deals with *FunctionSpec, not *Function. 

[[file:~/code/pulumi/pkg/codegen/schema/schema.go::func unmarshalFunctionSpec(funcSpec *FunctionSpec, data map\[string\]RawMessage) error {]]
https://github.com/pulumi/pulumi/blob/21770edc130518e2faa0c0506d60bb2af2997434/pkg/codegen/schema/schema.go#L1646-#L1646

Now *FunctionSpec doesn't have any changes.

Reply:
I'm assuming Function is only produced from FunctionSpec via bindFunctionDef, then we should have all construction paths covered.

Reply:
> Can you elaborate? The tests are verifying it working end to end.

The tests aren't really end to end at the moment, "gen-sdk" does a roundtrip via json marshaling to send schemas over grpc that you won't be seeing with the standard codegen tests.

Look at pkg/codegen/schema/schema.go around line 1176, you'll see you need to slot in pretty much the inverse of what you've done in bindFunctionDef.

Reply:
Ok, I see. I'll look into it.

Reply:
Thank you, edited pkg/codegen/schema/schema.go and threw in a test.